File: src/main/java/me/ztowne13/customcrates/listeners/DamageListener.java
package me.ztowne13.customcrates.listeners;

import me.ztowne13.customcrates.SpecializedCrates;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.EntityDamageByEntityEvent;

public class DamageListener implements Listener {
    SpecializedCrates sc;

    public DamageListener(SpecializedCrates sc) {
        this.sc = sc;
    }

    @EventHandler
    public void onFireworkExplode(EntityDamageByEntityEvent event) {
        if (event.getDamager().getCustomName() != null) {
            if (event.getDamager().getCustomName().equalsIgnoreCase("scf")) {
                if (!event.getDamager().isCustomNameVisible()) {
                    event.setCancelled(true);
                }
            }
        }
    }
}

File: src/main/kotlin/me/ztowne13/customcrates/listeners/DamageListener.kt
package me.ztowne13.customcrates.listeners

import org.bukkit.event.entity.EntityDamageByEntityEvent
import me.ztowne13.customcrates.SpecialisedCrates
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener

class DamageListener(var sc: SpecialisedCrates) : Listener {
    @EventHandler
    fun onFireworkExplode(event: EntityDamageByEntityEvent) {
        if (event.damager.customName != null) {
            if (event.damager.customName.equals("scf", ignoreCase = true)) {
                if (!event.damager.isCustomNameVisible) {
                    event.isCancelled = true
                }
            }
        }
    }
}